watertap/unit_models/tests/unit_test_harness.py

```diff
@@ -0,0 +1,175 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pytest
+
+from pyomo.environ import Block, assert_optimal_termination, value
+from pyomo.util.check_units import assert_units_consistent
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+)
+from idaes.core.solvers import get_solver
+from idaes.core.util.exceptions import InitializationError
+import idaes.core.util.scaling as iscale
+
+
+# -----------------------------------------------------------------------------
+class UnitAttributeError(AttributeError):
+    """
+    WaterTAP exception for generic attribute errors arising from unit model testing.
+    """
+
+
+class UnitValueError(ValueError):
+    """
+    WaterTAP exception for generic value errors arising from unit model testing.
+    """
+
+
+class UnitRuntimeError(RuntimeError):
+    """
+    WaterTAP exception for generic runtime errors arising from unit model testing.
+    """
+
+
+class UnitTestHarness:
+    def configure_class(self):
+        self.solver = None  # string for solver, if None use WaterTAP default
+        self.optarg = (
+            None  # dictionary for solver options, if None use WaterTAP default
+        )
+
```

Comment:
```suggestion

    # same as pytest.approx: can be overwritten by `configure()`
    self.default_absolute_tolerance = 1e-12
    self.default_relative_tolerance = 1e-06
    
```

Reply:
I would also include the comment that Ben suggested: `# same as pytest.approx: can be overwritten by `configure()``